File: 프로그래머스/lv0/120808. 분수의 덧셈/분수의 덧셈.java
class Solution {
    public int[] solution(int numer1, int denom1, int numer2, int denom2) {
        int[] answer = new int[2];
        
        int numer = numer1 * denom2 + numer2 * denom1;
        int denom = denom1 * denom2;
        
        int max = 1;
        for(int i=1; i<=numer && i<=denom; i++) {
            if(numer % i == 0 && denom % i ==0) {
                max = i;
            }
        }
        
        answer[0] = numer/max;
        answer[1] = denom/max;
        
        return answer;
    }
}

File: 프로그래머스/unrated/120808. 분수의 덧셈/분수의 덧셈.js
function solution(numer1, denom1, numer2, denom2) {
    const answer = [];
    let numer = numer1 * denom2 + numer2 * denom1;
    let denom = denom1 * denom2;
    
    let max;
    for(let i = 1; i <= numer && i <= denom; i++) {
        if(numer % i === 0 && denom % i === 0) max = i;
    }
    
    answer.push(numer/max)
    answer.push(denom/max)
    
    return answer;
}